### PR TITLE
Screen space clipping

### PR DESCRIPTION
**Pull request by MPanknin**
Initial support for clipping of points in screen space. Which parts to clip/discard and what other parts of a point cloud to render is defined as a rectangular region in normalized device coordinates on a per material basis. This means multiple datasets can be rendered and displayed in a side-by-side fashion by defining the appropriate rectangular region of its material.

Clipping can be performed horizontally as well as vertically. 

quad_lion

### CODE REVIEW

File: src/materials/shaders/pointcloud.frag [hunk 1]
```diff
@@ -61,6 +65,20 @@ void main() {
 	vec3 color = vColor;
 	float depth = gl_FragCoord.z;
 
+	#if defined (clip_horizontally) || defined (clip_vertically)
+		vec2 ndc = vec2((gl_FragCoord.x / screenWidth), (gl_FragCoord.y / screenHeight));
+
+		if(!(ndc.x > clipExtent.x && ndc.x < clipExtent.y))
+		{
+			discard;
```
Comment (fahadm) [inline src/materials/shaders/pointcloud.frag:73]: We already have a way of discarding points ( maybe take a look at the normal filtering part of the code).
 
If I am not mistaken, if we discard points like this, our point picking will still see these invisible points.

**Review by fahadm — CHANGES_REQUESTED** [review]

**Comment by fahadm** [comment]
The changes look quite good already, maybe take a look at my comment and see if it makes sense ( or comment if it doesn't)

**Comment by MPanknin** [comment]
Yes that's correct. It's four times the same dataset for illustration purposes. Later we will use this on different datasets or for comparing a point cloud against its mesh representation for example.

File: src/materials/shaders/pointcloud.frag (hunk 1 above)
Comment (MPanknin) [reply src/materials/shaders/pointcloud.frag:73]: Good point. I'll check if that would potentially cause problems for our use case. Thx!

**Pull request closed without merging**